server/services/history.py

```
@@ -1,32 +1,133 @@
+                        skipped += 1
+        except OSError as exc:
+            logger.warning("danmu history: cannot read %s: %s", HISTORY_FILE, exc)
+            return
+        logger.info(
+            "danmu history: restored %d records from disk%s",
+            loaded,
+            f" ({skipped} unreadable lines skipped)" if skipped else "",
+        )
+
+    def _try_rotate(self) -> None:
+        """超過上限就轉存成 .1（只留一份備份，跟 audit.log 一致）。"""
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)